src/process-cache.js

```diff
@@ -1,129 +1,61 @@
     log: ILog,
 ): Promise<ErrorCode> {
     const {autoFix, json} = options;
     const violationFileNames: Array<string> = [];
+    const jsonItems: Array<JsonItem> = [];
 
-    if (json) {
-        const jsonItems: Array<JsonItem> = [];
-        for (const file of Object.keys(cache)) {
-            try {
-                validateAndJson(options, file, cache, jsonItems);
-            } catch (e) {
-                log.error(
-                    `${Format.cwdFilePath(
-                        cwdRelativePath(file),
-                    )} update encountered error: ${e.message}`,
-                );
-            }
+    const getValidator = (): Validator => {
+        if (options.json) {
+            return async (options, file, cache) => {
+                jsonItems.push(...validateAndJson(options, file, cache));
+            };
         }
-        log.log(
-            JSON.stringify(
-                {
-                    version: version,
-                    rerunCommand: process.argv.join(" "),
-                    items: jsonItems,
-                },
-                null,
-                4,
-            ),
-        );
-        return ErrorCodes.SUCCESS;
-    } else {
         const fileValidator = autoFix ? validateAndFix : validateAndReport;
-        for (const file of Object.keys(cache)) {
-            try {
-                if (!(await fileValidator(options, file, cache, log))) {
-                    violationFileNames.push(file);
-                }
-            } catch (e) {
-                log.error(
-                    `${Format.cwdFilePath(
-                        cwdRelativePath(file),
-                    )} update encountered error: ${e.message}`,
-                );
-            }
-        }
-    }
-
-    if (violationFileNames.length > 0) {
-        const outputRerunCommand = () => {
-            log.log("");
-            /**
-             * There are some arguments we need to include to ensure that the launch
-             * string we output would work.
-             *
-             * We don't care about ignore files, because we're giving a list of specific
-             * files, but we do care about:
-             *  - comment matchers
-             *  - the root marker
-             */
-            const updateCommandParts = [];
-            updateCommandParts.push(getLaunchString());
-            const commentsArg = options.comments.sort().join(",");
-            if (commentsArg !== defaultArgs.comments) {
-                updateCommandParts.push("-c");
-                updateCommandParts.push(`"${commentsArg}"`);
-            }
-            const rootMarker = options.rootMarker;
-            if (rootMarker != null) {
-                updateCommandParts.push("-m");
-                updateCommandParts.push(`"${rootMarker}"`);
+        return async (options, file, cache) => {
+            if (!(await fileValidator(options, file, cache, log))) {
+                violationFileNames.push(file);
```

Comment:
I don't understand why there are still two paths here for this.

I would expect it this to just return one of the three validators and then have the resulting list of file names produced in the loop as before, which would be passed to the output functions for them to decide what to do. That way it's a single path through.